src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -280,21 +281,19 @@ export default function (props: BloodPressureVisualizationProps) {
         return Category.Unknown;
     }
 
-    function pageWeek(newStart: Date) {
-        setStartOfWeek(newStart);
-    }
-
-    useInitializeView(initialize, [], [props.previewState]);
+    useInitializeView(initialize, [], [props.previewState, props.weekStartsOn]);
 
     if (!bloodPressureData) {
-        return <LoadingIndicator innerRef={props.innerRef}/>;
+        return <LoadingIndicator innerRef={props.innerRef} />;
     }
     else {
         return (
             <div ref={props.innerRef}>
-                <DateRangeNavigator intervalType="Week" intervalStart={datePagerStartDate} onIntervalChange={pageWeek}></DateRangeNavigator>
-                {pageWeeklyData(datePagerStartDate)}
-                {pageWeeklyMetrics(datePagerStartDate)}
+                {(!dateRangeContext || dateRangeContext?.intervalType === "Week") && pageWeeklyData(intervalStart)}
+                {(!dateRangeContext || dateRangeContext?.intervalType === "Week") && pageWeeklyMetrics(intervalStart)}
+                {dateRangeContext && dateRangeContext?.intervalType !== "Week" &&
+                    <div className="mdhui-blood-pressure-interval-not-supported" ref={props.innerRef}>Blood Pressure Visualization does not support month view at this time</div>
```

Comment:
I would move this check way up to where you get the `dateRangeContext` in the first place.  If this component won't be able to render due to a context configuration setting, there is no reason for any data loading or other computations to take place.